sdk/identity/identity/src/msal/nodeFlows/msalNodeCommon.ts

```diff
@@ -174,6 +183,47 @@ export abstract class MsalNode extends MsalBaseUtilities implements MsalFlow {
       },
     };
   }
+  protected getApp(
+    appType: "publicFirst" | "confidentialFirst",
+    enableCae?: Boolean
+  ): msalNode.ConfidentialClientApplication | msalNode.PublicClientApplication;
+  protected getApp(appType: "public", enableCae?: Boolean): msalNode.PublicClientApplication;
+
+  protected getApp(
+    appType: "confidential",
+    enableCae?: Boolean
+  ): msalNode.ConfidentialClientApplication;
+
+  protected getApp(
+    appType: AppType,
+    enableCae?: Boolean
+  ): msalNode.ConfidentialClientApplication | msalNode.PublicClientApplication {
+    if (enableCae) {
```

Comment:
if we create a small type to hold both types of app for CAE and non-CAE we can simplify our logic a bit.

Assuming the following members:

```
private app: {
    public?: msalNode.PublicClientApplication;
    confidential?: msalNode.ConfidentialClientApplication;
  } = {};
  private caeApp: {
    public?: msalNode.PublicClientApplication;
    confidential?: msalNode.ConfidentialClientApplication;
  } = {};
```

We can change the body of this method to be something simpler like

```
const app = enableCae ? this.caeApp : this.app;
    if (appType === "publicFirst") {
      return (app.public ?? app.confidential)!;
    } else if (appType === "confidentialFirst") {
      return (app.confidential ?? app.public)!;
    } else if (appType === "confidential") {
      return app.confidential!;
    } else {
      return app.public!;
    }
```

there are similar efficiencies in the init method

Reply:
For init method, we won't get much out of the refactoring, since we want to instantiate the applications and assign it correctly. If we assign the new instantiated apps to just the `const app` it won't be reflected as a member of the class.